infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java

```diff
@@ -2,14 +2,26 @@
 
 import java.util.Objects;
 
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.redhat.service.smartevents.infra.api.APIConstants;
 import com.redhat.service.smartevents.infra.exceptions.BridgeError;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "Error",
+        allOf = { BaseResponse.class })
 public class ErrorResponse extends BaseResponse {
 
+    // https://issues.redhat.com/browse/MGDOBR-935
+    @JsonProperty("id")
+    private String id;
+
+    // https://issues.redhat.com/browse/MGDOBR-935
+    @JsonProperty("href")
+    private String href;
```

Comment:
As far as I see [`BaseResponse`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java#L9) also has `kind` and `name` as fields. If I understand the bug correctly you are duplicating fields from parent class as workaround. But why not also the other two fields?

Reply:
good question, I think it is just due to the validation bug. If we have a look on the openapi:
```
    Error:
      type: object
      allOf:
      - $ref: '#/components/schemas/ObjectReference'
      properties:
        id:
          type: string
        href:
          type: string
        code:
          type: string
        reason:
          type: string
```

We can seen there `allOf` referring to the parent `ObjectReference`. So I believe the SDK will be generated correctly, I mean `Error` model object for the SDK.

@dimakis @wtrocki could you please confirm my thoughts?

Reply:
yes @danielezonca, I believe you are correct with your thoughts here. @wtrocki may have some more insight but from my understanding that is accurate

Reply:
Generators require all properties to be part of the allOff. Mike did fix the same problem in this PR: 
https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/pull/204/files#diff-28a7ff5a0ccb4c306af5199105457eb5c1547945322341cdcac828b585e09f5c

Reapplying solution from that PR would give us much nicer result schema (and generally set good pattern for object inheritance in the future)

Reply:
What I saw in the openapi client generator earlier is that the `allOf` with only the parent class results in Java code properly having `Bean extends Parent` definitions. However, it results in invalid JavaScript clients. 

Using `allOf` with _both_ parent and child, the resulting Java puts everything into a single class (not actual inheritance at the Java level), but results in JavaScript having constructs like `export type ConsumerGroup = ConsumerGroupAllOf & ObjectReference;`. Personally I find the naming of the `ConsumerGroupAllOf` object confusing since it only includes the first `allOf` entry's properties.

Bottom line is that I think there is a lot of room for improvement in the code generator's handling of this both with consistency between languages and creating less "stuff" in the output. The Java output, for example, generates several "AllOf" classes that are not referenced elsewhere.